pkg/kubelet/util/manager/manager.go

```diff
@@ -49,12 +53,14 @@ type Store interface {
 	// AddReference adds a reference to the object to the store.
 	// Note that multiple additions to the store has to be allowed
 	// in the implementations and effectively treated as refcounted.
-	AddReference(namespace, name string)
+	AddReference(namespace, name string, podUID types.UID)
 	// DeleteReference deletes reference to the object from the store.
 	// Note that object should be deleted only when there was a
 	// corresponding Delete call for each of Add calls (effectively
 	// when refcount was reduced to zero).
-	DeleteReference(namespace, name string)
+	DeleteReference(namespace, name string, podUID types.UID)
 	// Get an object from a store.
 	Get(namespace, name string) (runtime.Object, error)
+	// GetPodsNeedSyncObjects gets pods that need syncing objects.
+	GetPodsNeedSyncObjects() []types.UID
```

Comment:
having a way to observe changes to namespace/name pairs seems ok, but making this interface pod-specific doesn't seem great... can the item → pod tracking be done as a separate layer?

Reply:
+1 - especially that we ideally would ideally like to promote this library to something more universally used - a usecases for it appeared in more places across the project.
@deads2k - FYI

Reply:
I think expanding `Store interface` with some methods may help this.
For example:
```
type object struct {
	namespace string
	name      string
	_type     string
	uid       types.UID
}

type key object
type value object
type valueWithChangedKeys struct {
	v           value
	changedKeys []key
}

type Store interface {
	AddReference(key, value)
	DeleteReference(key, value)
	Get(key) (runtime.Object, error)
	AddChangedKey(key)
	ChangedKeys() []key
	// Get all values for which the key has been changed, and the list of changed keys for each value.
	GetValuesWithChangedKeys() []valueWithChangedKeys
}

```

Reply:
Are we gonna introduce this "change observer" recently?